tiledb/sm/query/sparse_global_order_reader.cc

```diff
@@ -0,0 +1,918 @@
+    for (unsigned int f = 0; f < fragment_num; f++) {
+      uint64_t t = 0;
+      auto tile_num = fragment_metadata_[f]->tile_num();
+      for (t = frag_tile_idx_[f].first; t < tile_num; t++) {
+        // Use the bit vector to only load tiles in the subarray.
+        auto idx = t / 8;
+        auto bit = 1 << (t % 8);
+        if (result_tile_set_[f][idx] & bit) {
+          bool budget_exceeded = false;
+          RETURN_NOT_OK(load_coord_tile(
+              dim_num, per_fragment_memory, f, t, domain, &budget_exceeded));
+          tiles_found = true;
+
+          if (budget_exceeded)
+            break;
+        }
+      }
+
+      all_tiles_loaded_[f] = t == tile_num;
+      done_loading_coord_tiles &= all_tiles_loaded_[f];
+    }
+  } else {
+    // Load as many tiles as the memory budget allows.
+    for (unsigned int f = 0; f < fragment_num; f++) {
+      uint64_t t = 0;
+      auto tile_num = fragment_metadata_[f]->tile_num();
+      for (t = frag_tile_idx_[f].first; t < tile_num; t++) {
+        bool budget_exceeded = false;
+        RETURN_NOT_OK(load_coord_tile(
+            dim_num, per_fragment_memory, f, t, domain, &budget_exceeded));
+        tiles_found = true;
+
+        if (budget_exceeded)
+          break;
+      }
+
+      all_tiles_loaded_[f] = t == tile_num;
+      done_loading_coord_tiles &= all_tiles_loaded_[f];
+    }
+  }
+
+  done_loading_coord_tiles_ = done_loading_coord_tiles;
+
+  // No tiles found, return.
+  if (!tiles_found) {
+    return Status::Ok();
+  }
+
+  // Populate the temp vector for the common functions to work without change.
+  tmp_result_tiles_.clear();
+  for (auto& frag_result_tile : result_tiles_)
+    for (auto& result_tile : frag_result_tile)
+      tmp_result_tiles_.push_back(&result_tile);
+
+  // Read and unfilter zipped coordinate tiles. Note that
+  // this will ignore fragments with a version >= 5. Unfilter
+  // with a nullptr `rcs_index` argument to bypass selective
+  // unfiltering.
+  RETURN_CANCEL_OR_ERROR(
+      read_coordinate_tiles({constants::coords}, tmp_result_tiles_));
+  RETURN_CANCEL_OR_ERROR(
+      unfilter_tiles(constants::coords, tmp_result_tiles_, nullptr));
+
+  // Read and unfilter unzipped coordinate tiles. Note that
+  // this will ignore fragments with a version < 5. Unfilter
+  // with a nullptr `rcs_index` argument to bypass selective
+  // unfiltering.
+  RETURN_CANCEL_OR_ERROR(read_coordinate_tiles(dim_names_, tmp_result_tiles_));
+  for (const auto& dim_name : dim_names_) {
+    RETURN_CANCEL_OR_ERROR(
+        unfilter_tiles(dim_name, tmp_result_tiles_, nullptr));
+  }
+
+  // Compute the result cell slabs with the loaded coordinate tiles.
+  if (array_schema_->cell_order() == Layout::HILBERT) {
+    std::vector<std::vector<uint64_t>> values(fragment_metadata_.size());
+    RETURN_CANCEL_OR_ERROR(compute_result_cell_slabs(
+        per_fragment_memory,
+        HilbertCmpReverse(array_schema_->domain(), &values)));
```

Comment:
Not sure why this is happening in reverse.

Reply:
For min heap, we talked offline about this.